apps/core/utils.py

```diff
@@ -155,3 +157,125 @@ def format_task_data(data: Any) -> str:
         return json.dumps(data, indent=2)
     except Exception:
         return str(data)
+
+
+def log_setting_change(user, setting_key: str, new_value, source: str, request=None):
+    """
+    Log a settings change to the database.
+
+    Returns:
+        Setting object if successful, None if failed
+    """
+
+    # List of sensitive settings that should be redacted
+    sensitive_settings = [
+        "SECRET_KEY",
+        "PASSWORD",
+        "DATABASES",
+        "JWT_SECRET",
+        "API_KEY",
+    ]
+
+    # Check if this setting is sensitive
+    is_sensitive = any(sensitive in setting_key.upper() for sensitive in sensitive_settings)
+
+    # Redact sensitive values
+    if is_sensitive:
+        new_value_to_store = "***REDACTED***"
+    else:
+        # Convert values to JSON strings for storage
+        try:
+            new_value_to_store = json.dumps(new_value) if new_value is not None else None
+        except (TypeError, ValueError):
+            # If we can't convert to JSON, just convert to string
+            new_value_to_store = str(new_value) if new_value is not None else None
+
+    # Get IP address from request if available
+    ip_address = None
+    if request:
+        # Try to get real IP (handles proxies)
+        x_forwarded_for = request.META.get("HTTP_X_FORWARDED_FOR")
+        ip_address = x_forwarded_for.split(",")[0].strip() if x_forwarded_for else request.META.get("REMOTE_ADDR")
+
+    try:
```

Comment:
```suggestion

```

Reply:
On line 200, I see separate sequential 
  try/except blocks, not nested ones.